pkgsettings/pkgsettings.py

```diff
@@ -26,6 +31,36 @@ def as_dict(self):
             result.update(item.as_dict())
         return result
 
+    def children(self):
+        """
+        Tries to return a generator of all settings objects in the chain, recursively.
+        This might not yield all settings objects, if they include
+        other settings objects not by using the children() call.
+        :return: generator of settings objects.
+        """
+        for settings in self._chain:
+            yield settings
```

Comment:
I think since we yield, it doesn't matter if we do BFS or DFS, so I think they can actually be merged. I'll merge them.